docs/architecture/blazor-for-web-forms-developers/security-authentication-authorization.md

```diff
@@ -1,15 +1,379 @@
+    }
+    else
+    {
+        app.UseExceptionHandler("/Error");
+        // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+        app.UseHsts();
+    }
+
+    app.UseHttpsRedirection();
+    app.UseStaticFiles();
+
+    app.UseRouting();
+
+    app.UseAuthentication();
+    app.UseAuthorization();
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.MapRazorPages();
+    });
+}
+```
+
+ASP.NET Identity does not configure anonymous or role-based access to locations from `Startup.cs`. You will need to migrate any location-specific authorization configuration data to filters in ASP.NET Core. Make note of which folders and pages will require such updates. You will make these changes in the next section.
+
+### Updating individual pages to use ASP.NET Core Identity abstractions
+
+In your ASP.NET Web Forms application, if you had web.config settings to deny access to certain pages or folders to anonymous users, you would migrate these by simply adding the `[Authorize]` attribute to such pages:
+
+```razor
+@attribute [Authorize]
+```
+
+If you further had denied access except to those users belonging to a certain role, you would likewise migrate this by adding an attribute specifying a role:
+
+```razor
+@attribute [Authorize(Roles ="administrators")]
+```
+
+Note that the `[Authorize]` attribute only works on `@page` components that are reached via the Blazor Router. The attribute does not work with child components, which should instead use `AuthorizeView`.
+
+If you have logic within page markup for determining whether to display some code to a certain user, you can replace this with the `AuthorizeView` component. The [AuthorizeView component](https://docs.microsoft.com/aspnet/core/blazor/security/?view=aspnetcore-3.1#authorizeview-component) selectively displays UI depending on whether the user is authorized to see it. It also exposes a `context` variable that can be used to access user information.
+
+```razor
+<AuthorizeView>
+    <Authorized>
+        <h1>Hello, @context.User.Identity.Name!</h1>
+        <p>You can only see this content if you are authenticated.</p>
+    </Authorized>
+    <NotAuthorized>
+        <h1>Authentication Failure!</h1>
+        <p>You are not signed in.</p>
+    </NotAuthorized>
+</AuthorizeView>
+```
+
+You can access authentication state within procedural logic by accessing the user from a `Task<AuthenticationState` configured with the `[CascadingParameter]` attribute. This will get you access to the user, which can let you determine if they are authenticated and if they belong to a particular role. If you need to evaluate a policy procedurally, you can inject an instance of the `IAuthorizationService` and calls the `AuthorizeAsync` method on it. The following sample code demonstrates how to get user information and allow an authorized user to perform a task restricted by the `content-editor` policy.
```

Comment:
```suggestion
You can access authentication state within procedural logic by accessing the user from a `Task<AuthenticationState>` configured with the `[CascadingParameter]` attribute. This will get you access to the user, which can let you determine if they are authenticated and if they belong to a particular role. If you need to evaluate a policy procedurally, you can inject an instance of the `IAuthorizationService` and calls the `AuthorizeAsync` method on it. The following sample code demonstrates how to get user information and allow an authorized user to perform a task restricted by the `content-editor` policy.
```

Reply:
The `AuthenticationState` does first need to be setup as a cascading value before it can be bound to a cascading parameter like this. That's typically done using the `CascadingAuthenticationState` component.